packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.test.js

```diff
@@ -128,101 +134,222 @@ describe('overrides plugin', () => {
+            path: './',
             request: REQUEST_TWO_PATH
-        }
-        ;({resolver, callback} = setupResolverAndCallback(
-            null,
-            testTwoRequestContext,
-            null,
-            {},
-            callback
-        ))
+        })
+
+        ;({resolver, callback} = setupResolverAndCallback())
+
         const _overridesResolver = new OverridesResolverPlugin(options)
         _overridesResolver.handleHook(testTwoRequestContext, {}, callback, resolver)
 
         expect(callback).toHaveBeenCalled()
-        expect(resolver.ensureHook).not.toHaveBeenCalled()
         expect(resolver.doResolve).not.toHaveBeenCalledWith()
     })
 
-    test('jsx base template files can be replaced by tsx files', () => {})
+    test('jsx base template files can be replaced by tsx files', () => {
+        const REQUEST_PATH = 'newExtension'
+        const REQUEST_BASE_EXTENSION = '.jsx'
+        const REQUEST_OVERRIDE_EXTENSION = '.tsx'
 
-    test('npm @namespaces resolve correctly', () => {})
+        const testRequestContext = createRequestContextWith({
+            path: path.join(
+                '.',
+                'node_modules',
+                EXTENDS_TARGET,
+                REQUEST_PATH,
+                REQUEST_BASE_EXTENSION
+            ),
+            request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+        })
 
-    test('overridesDir is normalized with leading slash and forward slashes', () => {})
+        let {resolver, callback} = setupResolverAndCallback()
 
-    test('projectDir is normalized with forward slashes', () => {})
+        const overridesResolver = new OverridesResolverPlugin(options)
+        overridesResolver.handleHook(testRequestContext, {}, callback, resolver)
 
-    test('a nested overrides folder path/to/overrides resolves correctly', () => {})
+        expect(callback).toHaveBeenCalled()
+        expect(resolver.doResolve).toHaveBeenCalledWith(
+            null,
+            createRequestContextWith({
+                path: path.join(REWRITE_DIR, REQUEST_PATH + REQUEST_OVERRIDE_EXTENSION),
+                request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+            }),
+            expect.anything(),
+            expect.anything(),
+            expect.anything()
+        )
+    })
+
+    test('overridesDir and projectDir are normalized with leading slash and forward slashes', () => {
+        // In this test, all inputs use \\ to simulate Windows file paths
+        const REQUEST_PATH = 'exists'
+        const REQUEST_EXTENSION = '.jsx'
+        const testRequestContext = createRequestContextWith({
+            path: '.\\node_modules\\' + EXTENDS_TARGET,
+            request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+        })
+
+        const {resolver, callback} = setupResolverAndCallback()
+
+        const windowsOptions = {
+            overridesDir: '\\overrides',
+            extends: ['@salesforce/retail-react-app'],
+            projectDir: `src\\configs\\webpack\\test`
+        }
```

Comment:
So the problem here is that the `OverridesResolverPlugin` might allow for this, but we don't want to send the wrong signal internally to other engineers coming to this later. In reality, the `configs/webpack/config.js` file can't accept Windows formats like this because the webpack plugins expect posix / unix `/` slashes and setting the options like this in `package.json` would be a problem (not for `OverridesResolverPlugin` but for webpack config in `configs/webpack/config.js`

Reply:
Let's add a comment in the code here about the fact that this is not considered a valid value in `package.json` as it doesn't accept Windows paths in the outer calling context in `configs/webacpk/config.js`

Reply:
A comment has been added above this section